packages/create-platformatic-db/src/index.mjs

```diff
@@ -0,0 +1,155 @@
+
+async function getDependencyVersion (dependencyName) {
+  const require = createRequire(import.meta.url)
+  const pathToPackageJson = pathJoin(dirname(require.resolve(dependencyName)), 'package.json')
+  const packageJsonFile = readFileSync(pathToPackageJson, 'utf-8')
+  const packageJson = JSON.parse(packageJsonFile)
+  return packageJson.version
+}
+
+const createPlatformaticDB = async () => {
+  const logger = pino(pretty({
+    translateTime: 'SYS:HH:MM:ss',
+    ignore: 'hostname,pid'
+  }))
+
+  const help = helpMe({
+    dir: join(import.meta.url, '..', 'help'),
+    ext: '.txt'
+  })
+
+  const args = parseArgs(process.argv.slice(2), {
+    default: {
+      hostname: '127.0.0.1',
+      port: 3042,
+      database: 'sqlite',
+      migrations: 'migrations',
+      types: true,
+      typescript: false,
+      help: false
+    },
+    alias: {
+      h: 'hostname',
+      p: 'port',
+      db: 'database',
+      m: 'migrations',
+      t: 'types',
+      ts: 'typescript'
+    },
+    boolean: ['types', 'typescript']
+  })
+
+  if (args.help) {
+    await help.toStdout()
+    process.exit(0)
+  }
+
+  const username = await getUsername()
+  const version = await getVersion()
+  const pkgManager = getPkgManager()
+  const greeting = username ? `Hello, ${username}` : 'Hello,'
+  await say(`${greeting} welcome to ${version ? `Platformatic ${version}!` : 'Platformatic!'}`)
+  await say('Let\'s start by creating a new project.')
+
+  const wizardOptions = await inquirer.prompt([{
+    type: 'input',
+    name: 'dir',
+    message: 'Where would you like to create your project?',
+    default: './my-api',
+    validate: validatePath
+  }, {
+    type: 'list',
+    name: 'migrations',
+    message: 'Do you want to create default migrations?',
+    default: true,
+    choices: [{ name: 'yes', value: true }, { name: 'no', value: false }]
+  }, {
+    type: 'list',
+    name: 'generatePlugin',
+    message: 'Do you want to create a plugin?',
+    default: false,
+    choices: [{ name: 'yes', value: true }, { name: 'no', value: false }]
+  }, {
+    when: answers => (answers.generatePlugin),
+    type: 'list',
+    name: 'useTypescript',
+    message: 'Do you want to use TypeScript?',
+    default: false,
+    choices: [{ name: 'yes', value: true }, { name: 'no', value: false }]
+  }])
+
+  // Create the project directory
+  const projectDir = resolve(process.cwd(), wizardOptions.dir)
+  mkdirSync(projectDir)
+
+  const params = [
+    '--hostname', args.hostname,
+    '--port', args.port,
+    '--database', args.database,
+    '--migrations', wizardOptions.migrations ? args.migrations : '',
+    '--plugin', wizardOptions.generatePlugin,
+    '--types', wizardOptions.generatePlugin, // we set this always to true if we want to generate a plugin
+    '--typescript', !!wizardOptions.useTypescript
+  ]
+  await execaNode(join(import.meta.url, '../node_modules/@platformatic/db/db.mjs'), ['init', ...params], { cwd: projectDir })
```

Comment:
Can we import the `init` function and call it instead of creating a subprocess?

Reply:
This was my initial implementation, BUT the `init` code (especially the typescript generation) assumes that everything happens in the current folder. So we can, but if we do so we need to change this in `init` passing the (optional) current directory. Not strong opinion on this, though, followed the "subprocess" path to have less changes in `init`.